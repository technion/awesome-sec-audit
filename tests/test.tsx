import * as React from 'react';
import { mount } from 'enzyme';
import { mountToJson } from 'enzyme-to-json';
import * as renderer from 'react-test-renderer';

//import * as renderer from 'react-test-renderer';
import Start from '../assets/awesome-sec-audit.tsx';

test('render displays correctly', () => {

    const component = mount(
            <Start />
    );

    let tree = mountToJson(component);
    expect(tree).toMatchSnapshot();

    component.find("button").simulate("click");

    tree = mountToJson(component);
    expect(tree).toMatchSnapshot();

    component.find("input").simulate("change", {target: {value: 30} });
    component.find("button").simulate("click");
    tree = mountToJson(component);
    expect(tree).toMatchSnapshot();

};
