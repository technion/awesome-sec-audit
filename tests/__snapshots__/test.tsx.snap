exports[`test render displays correctly 1`] = `
<Start>
  <div>
    <button
      onClick={[Function]}>
      Click to get started!
    </button>
  </div>
</Start>
`;

exports[`test render displays correctly 2`] = `
<Start>
  <div>
    <input />
    <button
      onClick={[Function]}>
      test2
    </button>
  </div>
</Start>
`;

exports[`test render displays correctly 3`] = `
<Start>
  <div />
</Start>
`;
